iphone/Classes/TiMediaVideoPlayerProxy.m

```diff
@@ -890,40 +907,77 @@ - (void)handleLoadStateChangeNotification:(NSNotification *)note
+- (void)handleNaturalSizeAvailableNotification:(NSNotification *)note
 {
-  if ([self _hasListeners:@"playbackstate"]) {
-    NSDictionary *event = [NSDictionary dictionaryWithObject:[self playbackState] forKey:@"playbackState"];
-    [self fireEvent:@"playbackstate" withObject:event];
+  if ([self _hasListeners:@"naturalsizeavailable"]) {
+    [self fireEvent:@"naturalsizeavailable"
+         withObject:@{
+           @"naturalSize" : @{
+             @"width" : NUMFLOAT(movie.videoBounds.size.width),
+             @"height" : NUMFLOAT(movie.videoBounds.size.height)
+           }
+         }];
   }
+}
+
+// iOS < 10
+- (void)handlePlaybackStateChangeNotification:(NSNotification *)note
+{
+  TiVideoPlayerPlaybackState oldState = _playbackState;
 
   switch ([[movie player] status]) {
   case AVPlayerStatusUnknown:
   case AVPlayerStatusFailed:
     playing = NO;
+    _playbackState = TiVideoPlayerPlaybackStateInterrupted;
     break;
   case AVPlayerStatusReadyToPlay:
     playing = ([[movie player] rate] == 1.0);
+    if (playing) {
+      _playbackState = TiVideoPlayerPlaybackStatePlaying;
+    } else if (movie.player.currentItem.duration.value == movie.player.currentItem.currentTime.value || !movie.player.currentItem.canStepBackward) {
+      _playbackState = TiVideoPlayerPlaybackStateStopped;
+    } else {
+      _playbackState = TiVideoPlayerPlaybackStatePaused;
+    }
     break;
   }
+
+  if ([self _hasListeners:@"playbackstate"] && oldState != _playbackState) {
+    NSDictionary *event = [NSDictionary dictionaryWithObject:[self playbackState] forKey:@"playbackState"];
+    [self fireEvent:@"playbackstate" withObject:event];
+  }
 }
 
-- (void)handleNaturalSizeAvailableNotification:(NSNotification *)note
+// iOS 10+
+- (void)handleTimeControlStatusNotification:(NSNotification *)note
 {
-  if ([self _hasListeners:@"naturalsizeavailable"]) {
-    [self fireEvent:@"naturalsizeavailable"
-         withObject:@{
-           @"naturalSize" : @{
-             @"width" : NUMFLOAT(movie.videoBounds.size.width),
-             @"height" : NUMFLOAT(movie.videoBounds.size.height)
-           }
-         }];
+  TiVideoPlayerPlaybackState oldState = _playbackState;
+  playing = movie.player.timeControlStatus == AVPlayerTimeControlStatusPlaying;
+
+  if (movie.player.timeControlStatus == AVPlayerTimeControlStatusPlaying) {
+    _playbackState = TiVideoPlayerPlaybackStatePlaying;
+  } else if (movie.player.timeControlStatus == AVPlayerTimeControlStatusPaused) {
+    if (movie.player.currentItem.duration.value == movie.player.currentItem.currentTime.value) {
+      _playbackState = TiVideoPlayerPlaybackStateStopped;
+    } else {
+      _playbackState = TiVideoPlayerPlaybackStatePaused;
+    }
+  } else if ([TiUtils boolValue:[loadProperties valueForKey:@"autoplay"]]) {
+    _playbackState = TiVideoPlayerPlaybackStatePlaying;
+  }
+
+  if ([self _hasListeners:@"playbackstate"] && oldState != _playbackState) {
```

Comment:
In iOS 10+, if we observe 'playbackstate' event listener like -
`
video.addEventListener('playbackstate', function(e){
  Ti.API.info(e.playbackState);
});`
Ti.Media.VIDEO_PLAYBACK_STATE_STOPPED is never fired. 
Reason which I found is _playbackState gets update before callback 'handleTimeControlStatusNotification: ' called.

Reply:
@vijaysingh-axway I addressed this by making it an Optional that is not pre-set during configuration. See https://github.com/appcelerator/titanium_mobile/pull/9699/commits/daa4a9dc04a8298b8b57608f5e5868f8d0f5d72e for details. Good catch! 🙃

Reply:
Problem still not resolved. If we add the 'playbackstate' event listener as I suggested in above comment, when player finished playing movie it should fire "TiVideoPlayerPlaybackStateStopped" in 'playbackstate' event.